src/game/resources.rs

```diff
@@ -1 +1,112 @@
+use bevy::prelude::*;
+use rand::{seq::SliceRandom, thread_rng};
 
+use super::{input::components::InputText, word_match::components::Word, WordComplexity};
+
+/// Collection of words to choose from at "random"
+///
+/// * Must always use as `ResMut<WordBank>` due to modification for word uniqueness
+#[derive(Resource)]
+pub struct WordBank {
+    pub easy: Vec<String>,
+    easy_ptr: usize,
+    pub med: Vec<String>,
+    med_ptr: usize,
+    pub hard: Vec<String>,
+    hard_ptr: usize,
+    pub extreme: Vec<String>,
+    extreme_ptr: usize,
+}
+
+pub trait RandomWord {
+    fn get_word(
+        &mut self,
+        complexity: WordComplexity,
+        word_query: &Query<&Word, (With<Word>, Without<InputText>)>,
+    ) -> String;
+}
+
+impl RandomWord for WordBank {
+    fn get_word(
```

Comment:
This method now takes in a borrowed query to get all words. Right now it just checks every single spawned word, regardless if its visible to the player

current solution 
```
// guaranteed to be unique
pub fn gimmie_word(
    mut words: ResMut<WordBank>,
    word_q: Query<&Word, (With<Word>, Without<InputText>)>,
) {
  let word = words.get_word(WordComplexity::Easy, &word_q);
  // ...
}
```
old solution where the caller checks each time
```
// loop until unique
pub fn gimmie_word(
    mut words: ResMut<WordBank>,
    word_q: Query<&Word, (With<Word>, Without<InputText>)>,
) {
    let mut word = words.get_word(WordComplexity::Easy);
    let mut dup = word_q.iter().any(|w| word.1 == word);
    while dup {
        word = words.get_word(WordComplexity::Easy);
        dup = word_q.iter().any(|w| word.1 == word);
    }
    // ...
}
```

Reply:
This also allows us to reshuffle when at the end of our list, so no patterns happen